### PR TITLE
updates to the longest streak game

### PR DESCRIPTION
**Pull request by jimip85**

### CODE REVIEW

**Review by jimip85 — COMMENTED** [review]
- further modified other components to increase stylistic uniformity
- reduced redundant tailwind commands
(A)
 image
(B) Integrated Tailwind animations 
image

File: components/longestStreak/GameHUD.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+} from "../../pages/api/games/longestStreak";
+import {
+  setStage,
+  STAGE,
+  reset,
+  initializeGame,
+} from "../../redux/longestStreakSlice";
+import { Button } from "../ui/Button";
+
+export default function GameHUD({ data, gameState, user }) {
```
Comment (vithushan19) [inline components/longestStreak/GameHUD.tsx:14]: make sure you understand why I moved this code into a different file, and how it benefits the codebase by extracting this component out.

File: components/longestStreak/MultiplicationBlock.tsx [hunk 2]
```diff
@@ -23,16 +23,16 @@ export const MultiplicationBlock: React.FC<MultiplicationBlockProps> = ({
     <div className="cursor-pointer">
       <div
 
-        className={`md:w-20 md:h-20 w-10 h-10 flex justify-center items-center md:border-2 bg-gradient-to-b hover:bg-purple-900 text-white md:px-3 px-2 md:pt-0 pt-2 text-justify md:font-bold text-sm md:text-xl md:border-b-4 border-b-2 rounded-lg active:border-b-2 cursor-pointer float-left
+        className={`md:w-20 md:h-20 w-10 h-10 flex justify-center items-center md:border-2 bg-gradient-to-b hover:bg-red-400 text-white md:px-3 px-2 md:pt-0 pt-2 text-justify md:font-bold text-sm md:text-xl md:border-b-4 border-b-2 rounded-lg active:border-b-2 cursor-pointer float-left
 
         ${
           blockState === BlockState.PLAYER_ONE_SELECTED
-            ? "bg-[#16acea]"
+            ? "bg-red-500 animate-pulse"
```
Comment (vithushan19) [inline components/longestStreak/MultiplicationBlock.tsx:30]: great animations!

File: components/longestStreak/Rules.tsx [hunk 3]
```diff
@@ -8,43 +8,46 @@ export interface RulesProps {
...
-            <span className="text-xl">"harder problems"</span>
-            or continue practicing at the{" "}
-            <span className="text-xl">"same level".</span>
-          </ol>
-          <ol className="justify-start">
-            3. {""}
-            {""}When all squares have been magically colored in, click
-            <span className="text-xl">"Winner"</span> to see who conquered the
-            board on this mission!
+    <div className="h-screen">
```
Comment (vithushan19) [inline components/longestStreak/Rules.tsx:11]: do you understand how parent child relationships work? do you understand how we needed to make the parent h-screen, and then the next child h-full to maximize the height of this game. Only when the game was full height were we able to vertically center the UI.

A good way to understand the bounding boxes of different components is to give it a coloured border or background just so we can visually see where different elements are being rendered.

**Review by vithushan19 — APPROVED** [review]
this looks good to me. you can click the green merge button to get your changes into the mathchamp website.

Comments:
- seems like you're comfortable using the tailwind colours to make things look nice
- would like to see that you understand how to manipulate css grids and flexboxes, I didn't see much of that in this PR
- nice job learning how to implement tailwind animations. would love to see if you can extend this knowledge to build a custom animation perhaps for the quiz?

nice work overall, excited to see how you build upon this knowledge

File: components/longestStreak/GameHUD.tsx (hunk 1 above)
Comment (jimip85) [reply components/longestStreak/GameHUD.tsx:14]: For sure, I'll revise it further.

**Pull request merged**